x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -116,7 +116,10 @@ const VulnerabilitiesContent = ({ dataView }: { dataView: DataView }) => {
   const slicedPage = usePageSlice(data?.page, pageIndex, pageSize);
 
   const invalidIndex = -1;
-  const selectedVulnerability = data?.page[urlQuery.vulnerabilityIndex];
+
+  const selectedVulnerability = useMemo(() => {
```

Comment:
can you explain why useMemo is needed here? if the calculation is O(1), I assume it is redundant overhead

Reply:
cc: @Omolola-Akinleye

Reply:
@kfirpeled Good point.`useMemo` is not needed here. It's over-optimization